File: TunnelProject/Opc/OpcItemKey.cs
namespace TunnelProject.Opc
{
    internal sealed class OpcItemKey
    {
        private int _serverHandle;
        private readonly OpcComm _opcComm; // Add a reference to an OpcComm instance  

        public int ClientHandle { get; set; }

        public string PlcTagName { get; set; }

        public string GroupName { get; set; }

        public int ServerHandle
        {
            get
            {
                if (_serverHandle < 0)
                    _serverHandle = _opcComm.GetItemServerHandle(GroupName, PlcTagName); // Use the instance of OpcComm  
                return _serverHandle;
            }
            set
            {
                _serverHandle = value;
            }
        }

        public object Value { get; set; }

        public OpcItemKey(int clientHandle, string groupName, string tagName, OpcComm opcComm) // Add OpcComm as a parameter  
        {
            this.ClientHandle = clientHandle;
            PlcTagName = tagName;
            this.GroupName = groupName;
            ServerHandle = -1;
            _opcComm = opcComm ?? throw new ArgumentNullException(nameof(opcComm)); // Ensure OpcComm is not null  
        }
    }
}

File: TSA_Rail_System_TunnelProject/Wdw_Sust.OpcDa.Core/OpcItemKey.vb
Friend NotInheritable Class OpcItemKey

    Private _serverHandle As Integer

    Public Sub New(clientHandle As Integer, groupName As String, tagName As String)
        Me.ClientHandle = clientHandle
        PlcTagName = tagName
        Me.GroupName = groupName
        ServerHandle = -1
    End Sub

    Public Property ClientHandle() As Integer

    Public Property GroupName() As String
    Public Property PlcTagName() As String

    Public Property ServerHandle() As Integer
        Get
            If _serverHandle < 0 Then _serverHandle = OpcComm.Instance.GetItemServerHandle(GroupName, PlcTagName)
            Return _serverHandle
        End Get
        Set(value As Integer)
            _serverHandle = value
        End Set
    End Property

    Public Property Value() As Object
End Class